WiserTaskScheduler/WiserTaskScheduler/Modules/GenerateFiles/Services/GenerateFileService.cs

```diff
@@ -141,6 +149,40 @@ private async Task<JObject> GenerateFile(GenerateFileModel generateFile, List<in
                     Html = body
                 };
                 pdfFile = await htmlToPdfConverterService.ConvertHtmlStringToPdfAsync(pdfSettings);
+                
+                // Merge PDF's to generated PDF
+                if (generateFile.Body.MergePdfs.Where(p => !String.IsNullOrEmpty(p.WiserItemId)).ToArray().Length > 0)
+                {
+                    // Make stream of byte array
+                    var stream = new MemoryStream(pdfFile.FileContents);
+                    
+                    //  Create the merge result PDF document
+                    var mergeResultPdfDocument = new Document(stream);
+
+                    // Automatically close the merged documents when the document resulted after merge is closed
+                    mergeResultPdfDocument.AutoCloseAppendedDocs = true;
+
+                    // Set license key received after purchase to use the converter in licensed mode
+                    // Leave it not set to use the converter in demo mode
+                    mergeResultPdfDocument.LicenseKey = GclSettings.Current.EvoPdfLicenseKey;
```

Comment:
Please don't use `GclSettings.Current`, that should only be used if there is no other choice (like in a static function). In this case you should be able to access the GCL settings normally without issue. The proper/normal way is to add `IOptions<GclSettings>` to the constructor, so that dependency injection handles everything properly, put that in a field and access that field when you need it. There are hundreds of examples of this in the WTS, GCL en Wiser.